src/glue/romio/glue_romio.c

```diff
@@ -96,8 +96,6 @@ void MPIR_Ext_cs_exit(void)
 void MPIR_Ext_cs_yield(void)
 {
     /* TODO: check whether the progress engine is blocked */
-    MPID_THREAD_CS_YIELD(GLOBAL, MPIR_THREAD_GLOBAL_ALLFUNC_MUTEX);
```

Comment:
Should this then be `MPL_thread_yield()` and squashed with "romio: let romio have its own thread-safety"?

Reply:
Let's pull @roblatham00 in :).

Rob, there are only two place call `MPIR_Ext_cs_yield`. `ad_iread_coll.c` and `ad_iwrite_coll.c`, both inside a progress loop polling `Grequest`'s `poll_fn`. Is there a case that the `poll_fn` can only proceed when other threads of the same process gets to run?

Reply:
Initially added in 6cae5cab8a6d3f99b1b2f0ba317fa9ed48b604bc.

Reply:
the ad_i{read,write}_coll.c routines support nonblocking collectives.  I didn't write that ( Sangmin did) so I can't say off the top of my head what the yield is trying to do. 

I did write the `poll_fn`: the rules for what you can do in a poll_fn are not strictly defined -- we only have my EuroMPI paper attempting to do so.  My intention for the `poll_fn` is, as name suggests, to never block.   "read a flag from a register" and "make a system call that will return immediately" are the two use cases I ever had in mind.

Reply:
I see. Now ROMIO does not take MPICH lock what-so-ever, it is not blocking other threads running mpich progress. I guess the concern here is in the user-thread (argobot) case, we may still need explicit yield. I guess adding `MPL_thread_yield()` here is OK.

Reply:
@roblatham00 Thanks. The concern here is not whether the poll_fn will block; the concern is whether it will progress. So the question is whether it can progress without the participation of other threads. But I guess the answer is no, guessing from the reason that yield was added in the first place. I'll add `MPL_thread_yield()`, which currently is a noop on linux, a sleep(0) on mac and windows, and I am pondering on making it a couple hundred nanoseconds of `sched_yield()` out-of ch4 necessity. So if you are concerned that you can't afford a, say, 1us delay in the loop, let us know :)

Reply:
I'll add back the lock yield as well. If it was concerned the other threads need progress, then the other threads must need able to call ROMIO functions.